sdk/compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -107,9 +112,10 @@ extractDocs extractOpts diagsLogger ideOpts fp = do
 -- | This is equivalent to Haddock’s Haddock.Interface.Create.collectDocs
 collectDocs :: [LHsDecl GhcPs] -> [DeclData]
 collectDocs ds
-    | (nextDocs, decl:ds') <- spanMaybe getNextOrPrevDoc ds
+    | (nextDocs, decl:ds') <- spanMaybe getNextOrPrevOrWarnDoc ds
```

Comment:
These names are pretty confusing, esp. considering we're accounting for forms like

```
-- ^ Malformed prevdoc BEFORE a decl
data MyDecl
-- ^ Correctly formed prevdoc
```

Could we improve the comment above the collectDocs function and this variable to be more clear what we're passing around?